test/test_linalg.py

```diff
@@ -1533,6 +1533,296 @@ def test_eig_out(self, device, dtype):
+
+            # sort PyTorch output
+            # torch.argsort doesn't work well with complex inputs, NumPy sorting on CPU is used instead
+            # RuntimeError: _th_sort not supported on CUDAType for ComplexDouble
+            # RuntimeError: "sorting_kernel_method_name" not implemented for 'ComplexDouble'
+            ind = np.argsort(actual[0].cpu().numpy(), axis=-1)[::-1]
+            actual_np = [x.cpu().numpy() for x in actual]
+            sorted_actual = (
+                np.take_along_axis(actual_np[0], ind, axis=-1),
+                np.take_along_axis(actual_np[1], ind[:, None], axis=-1))
+
+            self.assertEqual(expected[0], sorted_actual[0])
+            self.assertEqual(abs(expected[1]), abs(sorted_actual[1]))
+
+        shapes = [(0, 0),  # Empty matrix
+                  (5, 5),  # Single matrix
+                  (0, 0, 0), (0, 5, 5),  # Zero batch dimension tensors
+                  (2, 5, 5),  # 3-dim tensors
+                  (2, 1, 5, 5)]  # 4-dim tensors
+        for shape in shapes:
+            run_test(shape)
+            run_test(shape, symmetric=True)
+
+    @skipCPUIfNoLapack
+    @skipCUDAIfNoMagma
+    @skipCUDAIfRocm
+    @dtypes(*floating_and_complex_types())
+    def test_eig_compare_backends(self, device, dtype):
+        def run_test(shape, *, symmetric=False):
+            from torch.testing._internal.common_utils import random_symmetric_matrix
+
+            if not dtype.is_complex and symmetric:
+                # for symmetric real-valued inputs eigenvalues and eigenvectors are also real-valued
+                a = random_symmetric_matrix(shape[-1], *shape[:-2], dtype=dtype, device=device)
+            else:
+                a = torch.randn(*shape, dtype=dtype, device=device)
+
+            actual = torch.linalg.eig(a)
+
+            if torch.cuda.is_available():
+                complementary_device = 'cpu' if self.device_type != 'cpu' else 'cuda'
+
+                # compare with CPU if dtype=='cuda' else with CUDA
+                expected = torch.linalg.eig(a.to(complementary_device))
+                self.assertEqual(expected[0], actual[0])
+                self.assertEqual(expected[1], actual[1])
+
+                # check out= variant
+                complex_dtype = dtype
+                if not dtype.is_complex and not symmetric:
+                    complex_dtype = torch.complex128 if dtype == torch.float64 else torch.complex64
+                out0 = torch.empty(0, dtype=complex_dtype, device=device)
+                out1 = torch.empty(0, dtype=complex_dtype, device=device)
+                ans = torch.linalg.eig(a, out=(out0, out1))
+                self.assertEqual(ans[0], out0)
+                self.assertEqual(ans[1], out1)
+                self.assertEqual(expected[0].to(complex_dtype), out0)
+                self.assertEqual(expected[1].to(complex_dtype), out1)
+
+                # check non-contiguous out
```

Comment:
Luckily these tests will be performed by the out= PR I mentioned so we won't have to keep writing them!

Reply:
Now that the out= PR is merged yes, I think we can remove these.